docker_cli/_shell_cmds.py

```diff
@@ -0,0 +1,369 @@
+            The target URL.
+        output_target : os.PathLike, optional
+            The file to redirect the URL contents to. If None, the URL contents
+            will go to stdout. Defaults to None.
+        stringify : bool, optional
+            Whether to return a string or list - if true, will return a string.
+            Defaults to False.
+
+        Returns
+        -------
+        Union[List[str], str]
+            The commands.
+        """
+        ...
+
+    @staticmethod
+    @abstractmethod
+    def get_name() -> str:
+        """
+        Returns the name of the command.
+
+        Returns
+        -------
+        str
+            The command name.
+        """
+        ...
+
+
+class Wget(URLReader):
+    @staticmethod
+    def generate_read_command(
+        target: str,
+        *,
+        output_target: Optional[Union[os.PathLike[str], str]] = None,
+        stringify: bool = False,
+    ) -> Union[List[str], str]:
+        retval = ["wget", target]
+        if output_target is not None:
+            retval += ["-O", str(output_target)]
+        if stringify:
+            return " ".join(retval)
+        return retval  # pragma: no cover
+
+    @staticmethod
+    def get_name() -> str:  # pragma: no cover
+        return "wget"
+
+
+class cURL(URLReader):
+    @staticmethod
+    def generate_read_command(
+        target: str,
+        *,
+        output_target: Optional[Union[os.PathLike[str], str]] = None,
+        stringify: bool = False,
+    ) -> Union[List[str], str]:
+        retval = ["curl", "--ssl", target]
+        if output_target is not None:
+            retval += ["-o", str(output_target)]
+        if stringify:
+            return " ".join(retval)
+        return retval  # pragma: no cover
+
+    @staticmethod
+    def get_name() -> str:  # pragma: no cover
+        return "curl"
+
+
+def get_package_manager(name: str) -> PackageManager:
+    """
+    Returns the PackageManager associated with a name.
+
+    Parameters
+    ----------
+    name : str
+        The name of the command
+
+    Returns
+    -------
+    PackageManager
+        An associated PackageManager object
+
+    Raises
+    ------
+    ValueError
+        When `name` does not correspond to a supported package manager.
+    """
+    name_cleaned = name.lower()
+    if name_cleaned == "apt-get":
+        return AptGet()
+    if name_cleaned == "yum":
+        return Yum()
+    else:
+        raise ValueError(f"Package manager '{name}' not recognized!")
+
+
+def get_url_reader(name: str) -> Type[URLReader]:
+    """
+    Returns the URLReader associated with a command name.
+
+    Parameters
+    ----------
+    name : str
+        The command name.
+
+    Returns
+    -------
+    URLReader
+        The associated URLReader class
+
+    Raises
+    ------
+    ValueError
+        When `name` does not correspond to a supported URL reader.
+    """
+    url_readers: Dict[str, Type[URLReader]] = {"curl": cURL, "wget": Wget}
```

Comment:
How come `get_package_manager()` returns a `PackageManager` *instance* but `get_url_reader()` returns a `URLReader` *class object*?

That is, how come you don't instantiate the classes like this:

```suggestion
    url_readers: Dict[str, URLReader] = {"curl": cURL(), "wget": Wget()}
```

Seems like it'd be preferable for these two to have similar interfaces & behavior so that users won't have to remember arbitrary inconsistencies.